### PR TITLE
autofix for 'sort-styles' rule

### PR DESCRIPTION
**Pull request by tanhauhau**
- added autofix for 'sort-styles' rule.

### CODE REVIEW

**Comment by maxis1718** [comment]
According to the [ESlint docs](https://eslint.org/docs/developer-guide/working-with-rules#applying-fixes), we need to specify the `fixable` property.

> Important: Without the fixable property, ESLint does not apply fixes even if the rule implements fix functions. Omit the fixable property if the rule is not fixable.

File: lib/util/stylesheet.js [hunk 1]
```diff
@@ -330,10 +357,13 @@ const astHelpers = {
   },
 
   hasArrayOfStyleReferences: function (node) {
-    return node && Boolean(
-      node.type === 'JSXExpressionContainer' &&
-      node.expression &&
-      node.expression.type === 'ArrayExpression'
+    return (
+      node &&
+      Boolean(
```
Comment (draperunner) [inline lib/util/stylesheet.js]: I guess this is just a Prettier/ESLint formatting fix, but how about moving the `Boolean` call up a level, like so:

```
return Boolean(
  node &&
  node.type === 'JSXExpressionContainer' &&
  node.expression &&
  node.expression.type === 'ArrayExpression'
)
```

**Comment by tanhauhau** [comment]
I can add some rest case for this..

File: lib/util/stylesheet.js (hunk 1 above)
Comment (tanhauhau) [reply lib/util/stylesheet.js]: i've reverted the formatting change, we probably should do this in another PR?

File: lib/util/stylesheet.js [hunk 2]
```diff
@@ -174,33 +212,26 @@ const astHelpers = {
...
     }
   },
 
   isStyleAttribute: function (node) {
     return Boolean(
       node.type === 'JSXAttribute' &&
-      node.name &&
-      node.name.name &&
-      node.name.name.toLowerCase().includes('style')
+        node.name &&
```
[the remark this answers is not in the document]
Comment (tanhauhau) [reply lib/util/stylesheet.js]: reverted formatting changes

**Comment by tanhauhau** [comment]
@draperunner [these test cases should suffice](https://github.com/Intellicode/eslint-plugin-react-native/blob/edd135379eac69380454c9f4af79d522af817fd9/tests/lib/rules/sort-styles.js#L379) ?

**Comment by tanhauhau** [comment]
added test case for shorthands too

**Comment by draperunner** [comment]
The test cases for shorthand properties look good. Great job!

Regarding comments, I think it's better to just not autofix a chunk if it contains comments. It's impossible to know what the comment means. Consider the case

```
const styles = StyleSheet.create({
  container: {
    d: 4,
    // Never EVER edit the next THREE lines!!!
    c: 3,
    a: 1,
    b: 2,
  }
})
```

If we just don't autofix this, but still report it as an error, the user will have to fix it manually. When we are not sure what the correct way to fix things is, it's better to ignore it than potentially break something.

**Comment by Intellicode** [comment]
Very nice! Sorry, I have been away for a bit, but I'll try to get out a release soonish!

**Comment by lukebars** [comment]
Any updates on this?

**Pull request merged**